core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -31,10 +34,12 @@
     private static final int INSTANCE_SIZE = toIntExact(ClassLayout.parseClass(UnnestingPositionsAppender.class).instanceSize());
```

Comment:
Rename this to `DictionaryAwarePositionsAppender`. We usually just have `DictionaryAwareXXX` which handles both RLE and dicts

Reply:
I would leave the name unchanged. Unnesting is a more important function of this class. Handling dictionaries is an additional feature.

Reply:
> Unnesting is a more important function of this class.

Unnesting became almost irrelevant at this point since now RLE and dicts cannot be nested